unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Scene/Main/Main.cs

```diff
@@ -112,5 +119,40 @@ protected virtual void OnDestroy()
         }
 
         private void OnGUI() { pluginSystem?.OnGUI(); }
+        protected virtual void InitializeSceneDependencies()
+        {
+            var bridges = Init("Bridges");
+            var mouseCatcher = Init("MouseCatcher").GetComponent<MouseCatcher>();
+            var environment = Init("Environment").GetComponent<EnvironmentReferences>();
+            var playerReferences = Init("Player").GetComponent<PlayerReferences>();
+
+            Init("HUDController");
+            Init("HUDAudioHandler");
+            Init("NavMap");
+            Init("SettingsController");
+
+            var sceneReferences = new SceneReferences(
```

Comment:
I see that this is initialized as a local instance, and the `SceneReferences` class fills out the static instance variable in its constructor. This escapes the usual idiom in which we do singletons and makes the singleton idiom pretty obscure. 

I think the intent is clearer if we just make `SceneReferences` inherit `Singleton<T>` and fill the references by just writing public fields or through a public `Initialize()` method.

Reply:
Done.